kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -55,6 +55,12 @@ def is_terminal(self, cterm: CTerm) -> bool:
             # <k> #halt ~> X:K </k>
             elif k_cell.arity == 2 and k_cell[0] == KEVM.halt() and type(k_cell[1]) is KVariable:
                 return True
+
+        program_cell = cterm.cell('PROGRAM_CELL')
+        # Fully symbolic program
+        if type(program_cell) is KVariable:
+            return True
```

Comment:
```suggestion
        program_cell = cterm.cell('PROGRAM_CELL')
        # Fully symbolic program
        if type(program_cell) is KVariable:
            if <k>-cell != <k> runLemma(...) </k> and <k>-cell != <k> runLemma(...) ~> _ </k>:
                return False
            return True
```

Reply:
I have implemented this now and tested locally, it appears to be working.